src/main/java/org/vaadin/addons/tatu/TabSheet.java

```diff
@@ -239,20 +249,58 @@ public void setSelected(String tab) {
         getElement().executeJs("this.setSelectedTab($0)", tab);
     }
 
+    /**
+     * Get selected tab index.
+     *
+     * @return Index of the tab, base 0 or -1 of no selection.
+     */
+    public int getSelectedIndex() {
+        return this.getElement().getProperty("selected", -1);
+    }
     /**
      * Set selected tab using index. This will fire TabChangeEvent. Sheet
      * attached to the tab will be shown.
      * 
      * @param index
      *            Index of the tab, base 0.
      */
-    public void setSelected(int index) {
+    public void setSelectedIndex(int index) {
         if (index < 0) {
             throw new IllegalArgumentException("Index can't be negative");
         }
         getElement().setProperty("selected", index);
     }
 
+    /**
+     * Get selected tab Component.
+     *
+     * @return Selected tab Component or <code>null</code> if no selection.
+     */
+    public Component getSelectedTab() {
+        String id = getSelected();
+        return id == null ? null : this.getComponent(id).orElse(null);
+    }
+    /**
+     * Set selected tab using index. This will fire TabChangeEvent. Sheet
+     * attached to the tab will be shown.
+     *
+     * @param tab
+     *            Component to select.
+     */
+    public void setSelectedTab(Component tab) {
```

Comment:
setSelectedByComponent ?

Reply:
See above. "Component" sounds good but "by" misleading. 

Overall, I think it is the best to think the naming use cases as three separate properties:
- String selected : the ID of the selected 
- int selectedIndex : Natural index of the tab
- Component selectedTab (or selectedComponent) : The tab content itself

What do you think, is the String needed for public API or could that be internal only? I think it would make sense in cases where storing component references is not possible (e.g. filling tabs in templates).

Reply:
Using setSelectedComponent